### DISCUSSION TITLE
Does avalonia support for embedding Windows Composition UI controls?

### DISCUSSION THREAD

**Discussion by WarWithinMe** [AvaloniaUI/Avalonia#11970] [General]
I successfully embed a Win32 version of WebView2 in my avalonia app.
And just find out that WebView2 supports Windows Composition UI.
According to doc, the WebView2 needs an `IDCompositionVisual` or 
a `Windows::UI::Composition::ContainerVisual` object to render.
And then implement some other methods to provide user interactions.

So does `IDCompositionVisual` or `Windows::UI::Composition::ContainerVisual` exists in Avalonia as a public API?

**Comment by maxkatz6** [comment]
Avalonia uses Windows::UI::Composition to render content into the window.
But in terms of embedding controls into the avalonia using NativeControlHost, you still need to get an HWND from somewhere. If WinUI version does have it, then it could work.

**Comment by WarWithinMe** [comment]
> Avalonia uses Windows::UI::Composition to render content into the window.

That's exactly what I'm trying to say. Since Avalonia already using Windows::UI::Composition,
so it might not need much effort to make Composition API available to the dev.

What I'm expecting is merely allowing us to create a Control that has a ContainerVisual object for rendering.
And that ContainerVisual has the Window's ContainerVisual as its parent.
Then we can just play around with Composition API to render things onto it.

I think the ContainerVisual might just be a canvas to draw on, and we will have to deal with the user input
the same way as a normal Avalonia Control.
Also, since other parts of the Avalonia App got flattened to a bitmap during rendering,
the ContainerVisual will always stay on top of other Avalonia Controls,
just like hosting with HWND, but this should be acceptable.

However, I didn't fully understand the whole Composition API, I might overlook the difficulty of this request.
